iroh-base/src/key.rs

```diff
@@ -334,40 +334,19 @@ impl SecretKey {
         self.secret.verifying_key().into()
     }
 
-    /// Generate a new [`SecretKey`] with the default randomness generator.
-    pub fn generate() -> Self {
-        let mut rng = rand::rngs::OsRng;
-        Self::generate_with_rng(&mut rng)
-    }
-
     /// Generate a new [`SecretKey`] with a randomness generator.
-    pub fn generate_with_rng<R: CryptoRngCore + ?Sized>(csprng: &mut R) -> Self {
-        let secret = SigningKey::generate(csprng);
+    ///
+    /// ```rust
+    /// // use the OsRng option for OS depedndent most secure RNG.
+    /// let mut rng = rand::rngs::OsRng;
+    /// let _key = iroh_base::SecretKey::generate(&mut rng);
```

Comment:
From your later changes seems like you can directly call `SecretKey::generate(rand::rngs::OsRng)` which might be a little simpler as recipe?

Reply:
you can, but that's not what most code will end up doing, which is why I made it two explicit steps